framework/src/executioners/EigenExecutionerBase.C

```diff
@@ -205,10 +199,15 @@ EigenExecutionerBase::inversePowerIteration(unsigned int min_iter,
   mooseAssert(tol_eig>0.0, "Invalid eigenvalue tolerance");
   mooseAssert(tol_x>0.0, "Invalid solution norm tolerance");
 
-  if ((_bx_execflag & (EXEC_TIMESTEP_END | EXEC_LINEAR)) == EXEC_NONE)
-    mooseError("rhs postprocessor for the power method has to be executed on timestep or residual");
-  if ((_xdiff_execflag & (EXEC_TIMESTEP_END | EXEC_LINEAR)) == EXEC_NONE)
-    mooseError("xdiff postprocessor for the power method has to be executed on timestep or residual");
+  // obtain the solution diff
+  const PostprocessorValue * solution_diff = NULL;
+  if (xdiff != "")
+  {
+    solution_diff = &getPostprocessorValueByName(xdiff);
+    ExecFlagType xdiff_execflag = _problem.getUserObject<UserObject>(xdiff).execBitFlags();
+    if ((xdiff_execflag & EXEC_LINEAR) == EXEC_NONE)
+      mooseError("solution diff postprocessor must be evaluated on linear currently");
```

Comment:
Thanks for the clarification.  I think if you just change the string to:
```
mooseError("The solution diff postprocessor requires execute_on = 'linear'");
```
or something similar, it will be more clear.